### PR TITLE
Reimplement percentile function in profile_analyzer.py

### PR DESCRIPTION
**Pull request by jtattermusch**
Context: b/76097549

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: tools/profiling/latency_profile/profile_analyzer.py [hunk 1]
```diff
@@ -184,24 +184,24 @@ def finish(self):
...
-    d1 = key(N[int(c)]) * (k - f)
-    return d0 + d1
+    idx = (len(N) - 1) * percent
+    idx_floor = math.floor(idx)
+    idx_ceil = math.ceil(idx)
+    if idx_floor != idx_ceil:
+        # interpolate the nearest element values
+        return (key(N[int(idx_floor)]) * (idx_ceil - idx) +
+                key(N[int(idx_ceil)]) * (idx - idx_floor))
+    return key(N[int(idx)])
```
Comment (vjpai) [inline tools/profiling/latency_profile/profile_analyzer.py]: You can eliminate idx_floor, idx_ceil, and the if condition altogether if you do:

return key(N[int(idx)]) + (idx - int(idx)) * (key(N[int(idx)+1]) - key(N[int(idx)]))

Comment (jtattermusch) [reply tools/profiling/latency_profile/profile_analyzer.py]: good point, but N[int(idx)+1]  can throw I think if percent=1.0

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by jtattermusch** [comment]
@vjpai   reimplemented the function PTAL.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Pull request merged**